chain/validation/applier.go

```diff
@@ -26,7 +25,7 @@ func (a *Applier) ApplyMessage(eCtx *vchain.ExecutionContext, state vstate.Wrapp
 	ctx := context.TODO()
 	st := state.(*StateWrapper)
 
-	base := cid.Undef // unused
+	base := st.Cid()
```

Comment:
`NewVM` does not accept an Undef CID here, we will need to start tracking the stateroot somehow.